go/private/bzl_format.bzl

```diff
@@ -0,0 +1,82 @@
+# Copyright 2016 The Bazel Go Rules Authors. All rights reserved.
+# Copyright 2016 The Closure Rules Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+""" Rules to support automatic formatting of bzl files
+"""
+
+def _bzl_format_yapf_repository_impl(ctx):
+  ctx.download_and_extract(
+      url = "https://codeload.github.com/google/yapf/zip/021238214778cf6bcce04dff3bef1f3f0da15530",
+      stripPrefix="yapf-021238214778cf6bcce04dff3bef1f3f0da15530",
+      type = "zip")
+  ctx.file("BUILD","""
+alias(
+    name="yapf",
+    actual="//yapf:yapf",
+    visibility = ["//visibility:public"],
+)
+""")
+  ctx.file("yapf/BUILD","""
+py_binary(
+    name="yapf",
+    srcs=glob(["**/*.py"]),
+    main="__main__.py",
+    visibility = ["//visibility:public"],
+)""")
+
+_bzl_format_yapf_repository = repository_rule(
+    implementation = _bzl_format_yapf_repository_impl,
+    attrs = {},
+)
+
+def bzl_format_repositories():
```

Comment:
I tried it, you are right that strip_prefix would work, but the resulting python code does not work unless it is in a directory called yapf....